runners/spark/src/test/java/org/apache/beam/runners/spark/SimpleWordCountTest.java

```diff
@@ -62,6 +71,29 @@ public void testRun() throws Exception {
     res.close();
   }
 
+  @Rule
+  public TemporaryFolder testFolder = new TemporaryFolder();
+
+  @Test
+  public void testOutputFile() throws Exception {
+    SparkPipelineOptions options = SparkPipelineOptionsFactory.create();
+    options.setRunner(SparkPipelineRunner.class);
+    Pipeline p = Pipeline.create(options);
+    PCollection<String> inputWords = p.apply(Create.of(WORDS)).setCoder(StringUtf8Coder
```

Comment:
Not particularly relevant to the content of your change, but the recommended way to set the coder here would be `Create.of(WORDS).withCoder(StringUtf8Coder.of())`.

Separately, it isn't entirely necessary: it is a bit of a hack, but the methods in the default `Create` implementation that [infer coders based on the values](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java#L277) are made public just so that runner overrides can still invoke them, [as the DataflowPipelineRunner does](https://github.com/apache/incubator-beam/blob/master/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java#L2508).

I hope the new whole-graph analysis will make it so individual runners no longer deal with this, but until then since it is not too hard to do, you might consider doing it for the Spark runner.